core/mutator.js

```diff
@@ -404,14 +406,20 @@ Blockly.Mutator.prototype.workspaceChanged_ = function(e) {
  * .viewWidth: Width of the visible rectangle,
  * .absoluteTop: Top-edge of view.
  * .absoluteLeft: Left-edge of view.
- * @return {!Object} Contains size and position metrics of mutator dialog's
- *     workspace.
+ * @return {!Blockly.utils.Metrics} Contains size and position metrics of
+ *     mutator dialog's workspace.
  * @private
  */
 Blockly.Mutator.prototype.getFlyoutMetrics_ = function() {
   return {
     viewHeight: this.workspaceHeight_,
     viewWidth: this.workspaceWidth_ - this.workspace_.getFlyout().getWidth(),
+    contentHeight: 0,
+    contentWidth: 0,
+    viewTop: 0,
+    viewLeft: 0,
+    contentTop: 0,
+    contentLeft: 0,
```

Comment:
Can we make these optional? Setting something like contentHeight to 0 seems a bit strange to me

Reply:
We could but they create a bigger headache elsewhere where we make assumptions that they exist, I will give it a go and see how bad it is.

Reply:
I looked into this, and I think it's not worth having to assert whether or not these values exist all of the place.

Reply:
Ok, could we add a comment explaining why we are setting these values to 0?

Reply:
Sounds good.

Reply:
The mutator workspace only uses a subset of Blockly.utils.Metrics as features such as scroll and zoom are unsupported.

Reply:
That looks great, thanks!